spec/catalog.yaml

```diff
@@ -396,6 +440,126 @@ components:
         location:
           type: string
 
+    Schema:
+      type: object
+      required:
+        - fields
+      properties:
+        metadata:
+          type: object
+          additionalProperties:
+            type: string
+        fields:
+          type: array
+          uniqueItems: true
+          items:
+            $ref: '#/components/schemas/Field'
+
+    Field:
+      type: object
+      required:
+        - type
+        - name
+        - id
+        - logicalType
+      properties:
+        type:
+          type: string
+          enum:
+            - PARENT
+            - REPEATED
+            - LEAF
+        name:
+          type: string
+        id:
+          type: int
+        parentId:
+          type: int
+        logicalType:
+          type: string
+          description: >
+            Logical types, currently support parameterized Arrow Type.
+            
+            PARENT types will always have logical type "struct".
+            
+            Logical type "map" is represented as a struct with a single child field "entries".
+            "entries" is a struct with fields "key" and "value".
+
+            REPEATED types may have logical types:
+            * "list"
+            * "large_list"
+            * "list.struct"
+            * "large_list.struct"
+            The final two are used if the list values are structs, and therefore the
+            field is both implicitly REPEATED and PARENT.
+          
+            LEAF types may have logical types:
+            * "null"
+            * "bool"
+            * "int8" / "uint8"
+            * "int16" / "uint16"
+            * "int32" / "uint32"
+            * "int64" / "uint64"
+            * "halffloat" / "float" / "double"
+            * "string" / "large_string"
+            * "binary" / "large_binary"
+            * "date32:day"
+            * "date64:ms"
+            * "decimal:128:{precision}:{scale}" / "decimal:256:{precision}:{scale}"
+            * "time:{unit}" / "timestamp:{unit}" / "duration:{unit}", where unit is "s", "ms", "us", "ns"
+            * "dict:{value_type}:{index_type}:false"
+        nullable:
+          type: boolean
+        metadata:
+          type: object
+          description: optional field metadata (e.g. extension type name/parameters)
+          additionalProperties:
+            type: string
+
+    WriterVersion:
```

Comment:
This is basically the guardrail that ensures that the server knows if there is any issue in creating a table with the user's environment. For example if the user library is too out of date, the server should reject the table creation request rather than creating a table that the client cannot consume.

Reply:
You mean the SDK version?

Reply:
I think it's more of the format version https://lancedb.github.io/lance/format.html#file-version